### ISSUE TITLE
AZ-301T04  Module 3 - azbb deployment is broken

### ISSUE THREAD

**Issue by thomasGusenleitner** [MicrosoftLearning/AZ-301-MicrosoftAzureArchitectDesign#94]
azbb -g $RESOURCE_GROUP_HUB_VNET -s $SUBSCRIPTION_ID -l $LOCATION -p ~/hub-vnet.json --deploy 

fails with:

templateViolation: 
'The template variable 'virtualMachineTemplate' is not valid: The language expression property 'templateLink' doesn't exist'

<img width="1902" alt="azbb-failure" src="https://user-images.githubusercontent.com/58548626/82046596-7edcd180-96b1-11ea-8f71-a0e408d2bc6b.png">

**Comment by polichtm** [comment]
Hi Thomas,
which exercise, task, and step are you referring to?
rgds
Marcin

**Comment by thomasGusenleitner** [comment]
Hi,

 

for example https://github.com/MicrosoftLearning/AZ-301-MicrosoftAzureArchitectDesign/blob/master/Instructions/AZ-301T04_Lab_Mod03_Deploying%20Network%20Infrastructure%20for%20use%20in%20Azure%20Solutions.md

 

there seems to be a general problem with azbb and execution local templates 

 

regards,

Thomas

 

Von: polichtm <notifications@github.com> 
Gesendet: Montag, 18. Mai 2020 03:29
An: MicrosoftLearning/AZ-301-MicrosoftAzureArchitectDesign <AZ-301-MicrosoftAzureArchitectDesign@noreply.github.com>
Cc: Gusenleitner Thomas <Thomas.Gusenleitner@voestalpine.com>; Author <author@noreply.github.com>
Betreff: EXT: Re: [MicrosoftLearning/AZ-301-MicrosoftAzureArchitectDesign] AZ-301T04 Module 3 - azbb deployment is broken (#94)

 

EXTERNAL SENDER

 

Hi Thomas,
which exercise, task, and step are you referring to?
rgds
Marcin

**Comment by polichtm** [comment]
Hi Thomas,
pls clarify which exercise, task, and step are you referring to?
rgds
Marcin

**Comment by thomasGusenleitner** [comment]
Hi Marcin,

https://github.com/MicrosoftLearning/AZ-301-MicrosoftAzureArchitectDesign/blob/master/Instructions/AZ-301T04_Lab_Mod03_Deploying%20Network%20Infrastructure%20for%20use%20in%20Azure%20Solutions.md

Exercise 1: Configure the lab environment
Task 5: Implement the hub component of the Hub ans Spoke design
Step 5: deploy the Hub-and-Spoke topology using azbb

Best regards,
Thomas



Von: polichtm <notifications@github.com>
Gesendet: Donnerstag, 21. Mai 2020 01:48
An: MicrosoftLearning/AZ-301-MicrosoftAzureArchitectDesign <AZ-301-MicrosoftAzureArchitectDesign@noreply.github.com>
Cc: Gusenleitner Thomas <Thomas.Gusenleitner@voestalpine.com>; Author <author@noreply.github.com>
Betreff: EXT: Re: [MicrosoftLearning/AZ-301-MicrosoftAzureArchitectDesign] AZ-301T04 Module 3 - azbb deployment is broken (#94)


EXTERNAL SENDER


Hi Thomas,
pls clarify which exercise, task, and step are you referring to?
rgds
Marcin

**Comment by polichtm** [comment]
Hi Thomas, 
As per https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/hybrid-networking/hub-spoke - unfortunately it appears that this particular pattern is no longer maintained. We'll need to remove it
rgds
Marcin

**Comment by elliotsegler** [comment]
This appears to be due to a bug in the azure CLI related to the deprecation of the `group deployment` command affecting how remote templates are validated. https://github.com/Azure/azure-cli/issues/12990

I'm not sure it needs to be moved, but it's not working while that bug exists.

**Comment by elliotsegler** [comment]
It's also worth mentioning that I tried this on my own Ubuntu box (i.e. without using cloud shell). The bug affects azure-cli 2.5.1, which is what cloud shell has. 2.6.0 has since been released and [doesn't appear to have that issue](https://github.com/MicrosoftDocs/azure-docs-cli/blob/9a443cb49352354c578139176f1cd707f18774f2/docs-ref-conceptual/release-notes-azure-cli.md#arm).

Maybe a sensible solution for the lab is to remove the dependency on cloud shell, or wait till MS update the versions of azure-cli that are available?

I've also noticed that this affects AZ-301T04_Lab_Mod02
